frontend/components/panels/BasePanel.tsx

```diff
@@ -0,0 +1,137 @@
+import Spinner from '@components/Spinner'
+import Tooltip from '@components/Tooltip'
+import { PythBalance } from '@pythnetwork/staking'
+import { useWallet } from '@solana/wallet-adapter-react'
+import { WalletModalButton } from '@solana/wallet-adapter-react-ui'
+import BN from 'bn.js'
+import { useState, ChangeEvent, useEffect } from 'react'
+
+export type BasePanelProps = {
+  description: string
+  isBalanceLoading: boolean
+  tokensLabel: string
+  balance: PythBalance | undefined
+
+  onAction: (amount: string) => void
+  isActionLoading: boolean | undefined
+  isActionDisabled: boolean | undefined
+  actionLabel: string
+}
+export function BasePanel({
```

Comment:
nitpick: imo this isn't a great abstraction because it's pretty inflexible. It's hard to customize the panels using this component to be slightly different from each other. 

I wouldn't worry about changing that in this PR though. if someone wants more customizability later, they can copy/paste this code into the other panel component.

Reply:
(a better abstraction would be more composable. An example here would be to make a component for the token input with the half/max buttons and the input text. If you had a couple of those little components, it would be easy to specify the LockPanel etc using a small number of lines of code, and it would also be more flexible in case you wanted to vary the appearances of those panels)

Reply:
I am going to address this in future. Prioritising other things over this.